Flank/src/test/java/com/walmart/otto/filter/TestFiltersTest.java

```diff
@@ -0,0 +1,127 @@
+package com.walmart.otto.filter;
+
+import static com.walmart.otto.filter.TestFilters.notIgnored;
+import static com.walmart.otto.filter.TestMethodFixtures.BAR_CLASSNAME;
+import static com.walmart.otto.filter.TestMethodFixtures.BAR_PACKAGE;
+import static com.walmart.otto.filter.TestMethodFixtures.FOO_CLASSNAME;
+import static com.walmart.otto.filter.TestMethodFixtures.FOO_PACKAGE;
+import static com.walmart.otto.filter.TestMethodFixtures.WITHOUT_FOO_ANNOTATION;
+import static com.walmart.otto.filter.TestMethodFixtures.WITHOUT_IGNORE_ANNOTATION;
+import static com.walmart.otto.filter.TestMethodFixtures.WITH_FOO_ANNOTATION;
+import static com.walmart.otto.filter.TestMethodFixtures.WITH_FOO_ANNOTATION_AND_PACKAGE;
+import static com.walmart.otto.filter.TestMethodFixtures.WITH_IGNORE_ANNOTATION;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import com.walmart.otto.testsupport.TestUtils;
+import java.nio.file.Path;
+import org.junit.Test;
+
+public class TestFiltersTest {
+
+  @Test
+  public void testFilteringByPackage() {
+    TestFilter filter = TestFilters.fromCommandLineArguments("package foo");
+
+    assertTrue(filter.shouldRun(FOO_PACKAGE));
+    assertFalse(filter.shouldRun(BAR_PACKAGE));
+  }
+
+  @Test
+  public void testFilteringByPackageNegative() {
+    TestFilter filter = TestFilters.fromCommandLineArguments("notPackage foo");
+
+    assertFalse(filter.shouldRun(FOO_PACKAGE));
+    assertTrue(filter.shouldRun(BAR_PACKAGE));
+  }
+
+  @Test
+  public void testFilteringByClassName() {
+    TestFilter filter = TestFilters.fromCommandLineArguments("class whatever.Foo");
+
+    assertTrue(filter.shouldRun(FOO_CLASSNAME));
+    assertFalse(filter.shouldRun(BAR_CLASSNAME));
+  }
+
+  @Test
+  public void testFilteringByClassNameNegative() {
+    TestFilter filter = TestFilters.fromCommandLineArguments("notClass whatever.Foo");
+
+    assertFalse(filter.shouldRun(FOO_CLASSNAME));
+    assertTrue(filter.shouldRun(BAR_CLASSNAME));
+  }
+
+  @Test
+  public void notIgnoredShouldFilterTestsWithTheIgnoreAnnotation() {
+    TestFilter filter = notIgnored();
+
+    assertFalse(filter.shouldRun(WITH_IGNORE_ANNOTATION));
+    assertTrue(filter.shouldRun(WITHOUT_IGNORE_ANNOTATION));
+  }
+
+  @Test
+  public void testFilteringByAnnotation() {
+    TestFilter filter = TestFilters.fromCommandLineArguments("annotation Foo,Bar");
+
+    assertTrue(filter.shouldRun(WITH_FOO_ANNOTATION));
+    assertFalse(filter.shouldRun(WITHOUT_FOO_ANNOTATION));
+  }
+
+  @Test
+  public void testFilteringByAnnotationWithSpaces() {
+    TestFilter filter = TestFilters.fromCommandLineArguments("annotation Foo, Bar");
```

Comment:
based on the `AndroidJUnitRunner` api this should fail, right?

Reply:
I not sure. However, we are lenient and we don't fail.

Reply:
ok, I think I see it in the code now. annotation `x, y` works because you are using `.anyMatch(annotations::contains)`. I think `package x, y` should fail because you are using startwith.